tests/bluetooth/tester/src/btp/btp_ots.h

```diff
@@ -7,11 +7,12 @@
  */
 
 #include <stdint.h>
+#include <zephyr/sys/util.h>
 
 /* OTS commands */
 #define BTP_OTS_READ_SUPPORTED_COMMANDS	0x01
 struct btp_ots_read_supported_commands_rp {
-	uint8_t data[0];
+	FLEXIBLE_ARRAY_DECLARE(uint8_t, data);
```

Comment:
Did you use a script to do `data[0]` changes? 

Below `name[0]`  is not using `FLEXIBLE_ARRAY_DECLARE`  :-)

Reply:
IIRC C++ does not allow empty structs (i.e. struct with only ` uint8_t data[];`, so that's why we need to use `FLEXIBLE_ARRAY_DECLARE` if a struct only contains a flexible array

Reply:
@cvinayak I see your point, unfortunately I don't see a better way to do it, at least not while we are supporting C++. Also, this macro is needed when using flexible arrays in unions.

There is always a risk of introducing issues for sure, but there are many benefits as well. Doing these changes I found some places where flexible arrays were not in the end of the structure and more importantly it will allow us to enable new compilation options to check some type out-of-bound accesses that is the majority cause of security issues.